prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-information-page/site-information-page.component.ts

```diff
@@ -83,4 +85,17 @@ export class SiteInformationPageComponent extends AbstractHealthAuthoritySiteReg
 
     this.routeUtils.routeRelativeTo(nextRoutePath);
   }
+
+  private addAsyncValidator() {
```

Comment:
Might be best to add this validation to the form state as it seems to be constant. That being said you can then get around and just create a AsyncValidatorFn and attach it when you create the form group. As well as adding a blur so its not firing all the time on change.

This is an example of what you can do
``` 
public buildForm(): void {
    this.formInstance = this.fb.group({
      siteName: ['', [Validators.required]],
      pec: [null, {
        asyncValidators: [this.pecValidator()],
        updateOn: 'blur'
      }],
      securityGroupCode: [null, [Validators.required]]
    });
  }

private pecValidator(): AsyncValidatorFn {
    return (control: AbstractControl): Observable<ValidationErrors | null> => {
      return this.healthAuthorityResource.checkPecExistsInOtherHEalthAuthority(
        this.healthAuthoritySiteService.site.healthAuthorityOrganizationId,
        control.value
      ).pipe(
        map((exists: boolean) => exists ? null : { duplicate: true })
      )
    }
  }
```

Reply:
I will do that I just have to import the HA resource service in the form file, I didn't do it initially just so I do not include the import in the form file.

Reply:
Updated, now the async validator is assigned at form building